From 6b6fd0411713a4f89b49e82ba23bb54f9a04ad84 Mon Sep 17 00:00:00 2001
From: Dmitry Petrochenko <dmitry.petrochenko@intel.com>
Date: Wed, 27 Mar 2013 13:58:28 +0700
Subject: Dalvik: Eliminate handleIfTaken pass as it is dead code now

BZ: 94861

Removed a handleIfTaken pass from compilerMIR2LIRJit since logic
of handleIfTaken is never applied if called from MIR2LIR.
The handleIfTaken function also removed since it is not used
anymore.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I8e5752abfc09fb230141a49b20986aac84ae8ebb
Orig-MCG-Change-Id: I3ee9cc1118565b71bcef2fc936b701f3024ba465
Signed-off-by: Dmitry Petrochenko <dmitry.petrochenko@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/codegen/x86/CodegenInterface.cpp |   59 +-------------------------
 1 files changed, 1 insertions(+), 58 deletions(-)

diff --git a/vm/compiler/codegen/x86/CodegenInterface.cpp b/vm/compiler/codegen/x86/CodegenInterface.cpp
index 135c7e9..58c5099 100644
--- a/vm/compiler/codegen/x86/CodegenInterface.cpp
+++ b/vm/compiler/codegen/x86/CodegenInterface.cpp
@@ -1433,53 +1433,6 @@ static void handleFallThroughBranch (CompilationUnit *cUnit, BasicBlock *bb, Bas
 }
 
 /**
- * @brief Handle the if back-end case where the taken branch has a table and we must merge
- * @param cUnit the CompilationUnit
- * @param bb the current BasicBlock
- */
-static void handleIfTaken (CompilationUnit *cUnit, BasicBlock *bb)
-{
-    BasicBlock *ft = bb->fallThrough;
-    BasicBlock *taken = bb->taken;
-
-    //Only care if we have a taken and a fallthrough
-    if (ft != 0 && taken != 0)
-    {
-        //Paranoid, we should have an if at the end
-        assert (bb->lastMIRInsn != 0 &&
-                bb->lastMIRInsn->dalvikInsn.opcode >= OP_IF_EQ &&
-                bb->lastMIRInsn->dalvikInsn.opcode <= OP_IF_LEZ);
-
-        //Now only care about it if the taken has a table
-        BasicBlock_O1 *takenO1 = reinterpret_cast<BasicBlock_O1 *> (taken);
-        assert (takenO1 != 0);
-
-        if (takenO1->associationTable.hasBeenFinalized () == true)
-        {
-            //Ok we need a BB between us and the taken
-            BasicBlock *inter = dvmCompilerNewBB (kDalvikByteCode, cUnit->numBlocks++);
-            dvmInsertGrowableList(&cUnit->blockList, (intptr_t) inter);
-
-            //Now update links
-            inter->fallThrough = taken;
-            bb->taken = inter;
-
-            //Update predecessors, the back-end must do it itself
-            if (taken->predecessors != 0)
-            {
-                dvmClearBit (taken->predecessors, bb->id);
-                dvmSetBit (taken->predecessors, inter->id);
-            }
-
-            if (inter->predecessors != 0)
-            {
-                dvmSetBit (inter->predecessors, bb->id);
-            }
-        }
-    }
-}
-
-/**
  * @brief Create a new record of 64bit constant in use
  * @details Allocates memory to store a 64bit constant and its details. All address fields
  *          are initialized to NULL.
@@ -1779,17 +1732,7 @@ static void compilerMIR2LIRJit(CompilationUnit *cUnit, JitTranslationInfo *info)
 
     if(gDvm.executionMode == kExecutionModeNcgO1) {
 
-        dvmGrowableListIteratorInit(&cUnit->blockList, &iterator);
-
-        //We do a first pass of the basic blocks before preprocessing because handleIfTaken might add some
-        for (bb = (BasicBlock *) (dvmGrowableListIteratorNext(&iterator));
-                bb != NULL;
-                bb = (BasicBlock *) (dvmGrowableListIteratorNext(&iterator))) {
-            //Handle the case of the taken if, in case an association table is already there
-            handleIfTaken (cUnit, bb);
-        }
-
-        //Start over the iterator
+        //Go over the basic blocks of the compilation unit
         dvmGrowableListIteratorInit(&cUnit->blockList, &iterator);
         for (bb = (BasicBlock *) (dvmGrowableListIteratorNext(&iterator));
                 bb != NULL;
-- 
1.7.4.1

